projects/step-frontend/src/lib/modules/timeseries/modules/_common/types/filter/filter-utils.ts

```diff
@@ -226,4 +226,45 @@ export class FilterUtils {
         throw new Error('Unrecognized type: ' + type);
     }
   }
+
+  public static createEmptyFilterTextItem(attribute: string): FilterBarItem {
+    const fieldType = this.getFilterFieldType(attribute);
+    return {
+      removable: true,
+      label: this.getFilterFieldLabel(attribute),
+      type: fieldType,
+      attributeName: attribute,
+      isLocked: false,
+    } as FilterBarItem;
+  }
+
+  private static getFilterFieldType(attribute: string): FilterBarItemType {
+    switch (attribute) {
+      case 'eId':
+        return FilterBarItemType.EXECUTION;
+      case 'taskId':
+        return FilterBarItemType.TASK;
+      case 'planId':
+        return FilterBarItemType.PLAN;
+      default:
+        return FilterBarItemType.FREE_TEXT;
+    }
+  }
```

Comment:
We are deciding on the type based on an attribute here, I would not call it "get" but "determineFilterFieldType", this would be a differentiation to the other function in the case you switch it to be based on the FilterBarItemType. (see comment there)

Reply:
agree!